plugins/inputs/windows_event_log/wineventlog/wineventlog.go

```diff
@@ -277,10 +265,14 @@ func (l *windowsEventLog) read() ([]*windowsEventLogRecord, error) {
 	}
 	// Decode the events into objects
 	if numRead == 0 {
-		return nil, nil
+		return nil
 	}
 
-	return l.getRecords(eventHandles[:numRead])
+	records := l.getRecords(eventHandles[:numRead])
+	if len(records) != int(numRead) {
+		log.Printf("E! [windows_event_log] requested events (%d) != returned events (%d)\n", numRead, len(records))
```

Comment:
I am leaning towards making this a DEBUG level message.
Please share your thoughts.

Reply:
I agree with debug

Reply:
Changed.